### DISCUSSION TITLE
How to properly add a tls-config in axum?

### DISCUSSION THREAD

**Discussion by Kimishu** [tokio-rs/axum#3094] [Q&A]
I was looking for a solution how to make my connection secured and I decided to add a tls for my axum app.
Then after following steps in examples ([this](https://github.com/tokio-rs/axum/blob/main/examples/tls-rustls/src/main.rs) and [this](https://github.com/tokio-rs/axum/blob/main/examples/tls-graceful-shutdown/src/main.rs)) I got CMakeError:
```
--- stderr
  CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
    Compatibility with CMake < 3.5 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.


  CMake Deprecation Warning at aws-lc/CMakeLists.txt:1 (cmake_minimum_required):
    Compatibility with CMake < 3.5 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.
```
As I see, something wrong with aws-lc.. Maybe somebody knows a solution?

### axum version

0.7.9

**Comment by mladedav** [comment]
Try `cargo tree -i aws-lc-...` to see the version and then look at the dependency if there are any known issues.

I suspect there will be issues with OpenSSL or another native dependency.

**Comment by jplatte** [comment]
The messages you shared are just warnings, there must be something else that is actually causing the build to fail. In any case, this seems unrelated to axum so you'll probably get better help elsewhere (if that library doesn't have its own support channel, StackOverflow and users.rust-lang.org come to mind).

**Comment by Kimishu** [comment]
@jplatte @mladedav 
Thanks for reply! I forgot to mention, this problem caused by `axum-server`.
I'll put all error log here.

(I tried to load it in raw, in empty project, it doesn't work too. As I remember, git contains ssl.. Maybe problem in there?)
I'll try to write to `axum-server`'s repo, but maybe u have any ideas what I can use now instead? (without waiting for fix)
[log.txt](https://github.com/user-attachments/files/18228280/log.txt)
